doc/sphinx/src/README.rst

```diff
@@ -178,6 +178,8 @@ Mesh
 ^^^^
 
 -  ``InitUserMeshData``
+-  ``ProblemGenerator``
+-  ``PostInitialization``
 -  ``PreStepUserWorkInLoop``
```

Comment:
While we're at it, I think in the context of this PR it'd be great to also add `UserWorkBeforeLoop` to the doc and emphasize the difference between `PostInit` and `UserWorkBeforeLoop` (as on first sight, they may seem identical in purpose).

Reply:
On it.